### ISSUE TITLE
in Eval.py line 25    pred = (pred > 0.5).float()

### ISSUE THREAD

**Issue by XUYUNYUN666** [milesial/Pytorch-UNet#107]
why? Can I remove it?

**Comment by milesial** [comment]
It's the threshold, why would you want to remove it ?

**Comment by XUYUNYUN666** [comment]
> It's the threshold, why would you want to remove it 


> It's the threshold, why would you want to remove it ?

Because if my dataset is multi-classes, the code is uncorrect?

**Comment by milesial** [comment]
if it is multiclass you may want to take the argmax, it depends on your need, if your problem is multilabel or not.

**Comment by ljw919** [comment]
> if it is multiclass you may want to take the argmax, it depends on your need, if your problem is multilabel or not.

Well, if i want to use it for multi-calssess, can i remove the threshold, just add the softmax and argmax(out,dim=0)? or maybe i just need to plot the outcome of softmax?Thank u.

**Comment by milesial** [comment]
Yes of course you would have to remove the threshold. Softmax is not necessary if all you are interested in is the argmax. You could plot the softmax if you need probabilities and not hard masks. Depends on your problem.

**Comment by ljw919** [comment]
Thank u for your replying. 😋So u mean that, if I want to get the true mask of the pic. I could directly add the argmax to the output without softmax? I once thought that argmax should be add with the probability of the output.Thank u!

**Comment by milesial** [comment]
Yes, argmax takes the max, and softmax is just for normalisation and does not change the order. Thus for prediction you can apply argmax on the raw logits without the need for probabilities
